### PR TITLE
feat(216): add new implement of git oauth

### LINKED ISSUE #216: A new implement of git OAuth is needed.
**What happened**
- The new cyclone API use `project` to manage `pipeline`, changes should be made to match it.

**What you expected to happen**
- A new implement of git OAuth.

**Reproduce steps** (as minimally and precisely as possible)
- [design API and data structure]
- [coding]

### PR DESCRIPTION
**Pull request by zz-bmj**
## Fixes #216

### CODE REVIEW

**Comment by zz-bmj** [comment]
@supereagle PTAL

File: pkg/server/router/router.go [hunk 1]
```diff
@@ -188,3 +197,36 @@ func (router *router) registerPipelineRecordAPIs(ws *restful.WebService) {
 		Param(ws.PathParameter("pipeline", "name of the pipeline").DataType("string")).
 		Param(ws.PathParameter("recordId", "id of the pipeline record").DataType("string")))
 }
+
+// registerScmAPIs registers scm related endpoints.
+func (router *router) registerScmAPIs(ws *restful.WebService) {
+	logdog.Info("Register scm APIs")
+
+	ws.Path(APIVersion).Consumes(restful.MIME_JSON).Produces(restful.MIME_JSON)
+	// GET /api/v1/projects/{project}/scm/{resource}/token
```
Comment (supereagle) [inline pkg/server/router/router.go]: What about resource -> type?  It is not a resource, but a scm type.

File: pkg/scm/provider/github.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	if err := g.setToken(code, state); err != nil {
+		return "", err
+	}
+
+	return redirectURL, nil
+}
+
+// setToken sets the token.
+func (g *GitHubManager) setToken(code, state string) error {
+
```
Comment (supereagle) [inline pkg/scm/provider/github.go]: Remove all empty lines before the code of functions, same errors in other places.

File: pkg/server/router/router.go (hunk 1 above)
Comment (zz-bmj) [reply pkg/server/router/router.go]: done

File: pkg/scm/provider/github.go (hunk 2 above)
Comment (zz-bmj) [reply pkg/scm/provider/github.go]: ok

**Comment by supereagle** [comment]
LGTM

**Pull request merged**